### PR TITLE
Update Mac Clang Docs

### PR DESCRIPTION
**Pull request by AlexandraKemperMS**
Updating Mac Clang docs with better screenshots, writing, files, and addressing the C++98 issue we had previously. Also updating the Mac config docs to be on par with the MinGW config docs.

### CODE REVIEW

**Comment by AlexandraKemperMS** [comment]
Hi @gregvanl  - While reviewing the [doc PR](https://github.com/microsoft/vscode-docs/pull/6467) created by @FlurinArner I saw a lot of little edits that needed to happen to these docs in general, so I incorporated FlurinArner's edits as well as added others to get the docs back up to date and on par with the MinGW config docs we have. Based on my run through, the C++98 problem has been resolved. Could you also review these edits?

File: docs/cpp/config-clang-mac.md [hunk 1]
```diff
@@ -156,25 +159,26 @@ Your new `tasks.json` file should look similar to the JSON below:
...
 
-The `command` setting specifies the program to run. In this case, `"clang++"` is the driver that causes the Clang compiler to expect C++ code and link against the C++ standard library.
+The `command` setting specifies the program to run. In this case, that is`clang++`.
 
-The `args` array specifies the command-line arguments that will be passed to clang++. These arguments must be specified in the order expected by the compiler.
+The `args` array specifies the command-line arguments that are passed to clang++. These arguments must be specified in the order expected by the compiler.
 
-This task tells the C++ compiler to compile the active file (`${file}`), and create an output file (`-o` switch) in the current directory (`${fileDirname}`) with the same name as the active file (`${fileBasenameNoExtension}`), resulting in `helloworld` for our example.
+This task tells the C++ compiler to take the active file (`${file}`), compile it, and create an output file (`-o` switch) in the current directory (`${fileDirname}`) with the same name as the active file but with the `.exe` extension (`${fileBasenameNoExtension}`). This process creates `helloworld`.
```
Comment (gregvanl) [inline docs/cpp/config-clang-mac.md:173]: Shouldn't this be:
"with the same name as the active file but without a file extension....

Comment (gregvanl) [reply docs/cpp/config-clang-mac.md:173]: I'll update this in a follow up PR

Comment (gregvanl) [reply docs/cpp/config-clang-mac.md:173]: Nope, I'll take care of it. I've got a follow up PR in progress fixing a few nits

**Pull request merged**